lib/resty/core/ctx.lua

```diff
@@ -53,6 +53,12 @@ local _M = {
 }
 
 
+-- use a new ctxs table to make LuaJIT JIT compiler happy to generate more
+-- efficient machine code.
+local ctxs = {}
+registry.ngx_lua_ctx_tables = ctxs
```

Comment:
```
    /* create the registry entry for the Lua request ctx data table */
    lua_pushliteral(L, ngx_http_lua_ctx_tables_key);
    lua_createtable(L, 0, 32 /* nrec */);
    lua_rawset(L, LUA_REGISTRYINDEX);
```
the original table was created in c, need to do something with it.

Reply:
Now, the only way to set `ngx.ctx` is using this `resty.core.ctx` module, no Lua C API way now.
So, it's safe enough now.